subsys/fs/fs.c

```diff
@@ -679,11 +673,29 @@ int fs_mount(struct fs_mount_t *mp)
 
 		if (strncmp(mp->mnt_point, itr->mnt_point, len) == 0) {
 			LOG_ERR("mount point already exists!!");
-			rc = -EBUSY;
-			goto mount_err;
+			return -EBUSY;
 		}
 	}
 
+	if (length) {
```

Comment:
`validate_mp` is internal function, so just require `length` to be not null with assert at the beginning, get rid of the check and make sure that `fs_mkfs` provides it with pointer, even if it discards the value.

Reply:
In `fs_mkfs` we don't need the length of a mount point. I decided to make the length argument optional. I think that making this argument required is worse solution (then we will have to pass pointer to some unused variable).

Reply:
This argument is passed whether you give here a pointer or NULL, which is pointer. 
You have just added additional, redundant, check in internal only function, which will be check for no reason, other than make it possible for the never omitted argument to be null.

Reply:
But why is creating an unused variable better? Here we have to make a choice of option that will be less confusing for someone who will read this in the future.

I think that passing a pointer to an unused variable will cause more confusion. The reader will spend some time wondering why we pass here a pointer to that variable. On the other hand if he sees a NULL pointer there, he will think “okay, we don’t care about the length of the mp”.